src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,101 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size == threshold) {
+            resize();
+        }
+        putValue(hash(key), key, value);
+    }
 
+    private void putValue(int hash, K key, V value) {
+        Node<K,V> newNode = new Node<>(hash, key, value);
+        if (table[getIndex(key)] == null) {
+            table[getIndex(key)] = newNode;
+        } else {
+            Node<K,V> tempNode = table[getIndex(key)];
+            while (tempNode != null) {
+                if (Objects.equals(tempNode.key, key)) {
+                    tempNode.value = value;
+                    return;
+                }
+                if (tempNode.next == null) {
+                    tempNode.next = newNode;
+                    break;
+                }
+                tempNode = tempNode.next;
+            }
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> tempNode = table[getIndex(key)];
+        while (tempNode != null) {
+            if (Objects.equals(tempNode.key, key)) {
+                return tempNode.value;
+            }
+            tempNode = tempNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        return hash(key) >= 0 ? hash(key) % table.length : -hash(key) % table.length;
```

Comment:
```suggestion
        return hash(key) >= 0 ? hash(key) % table.length : hash(key) % table.length;
```
better make hash(key) separate  variable